site/docs/row-level-deletes.md

```diff
@@ -0,0 +1,196 @@
+
+t2: data.1 data.2 data.3 delete.3_1    
+                         delete.3_2 (delete)
+```
+
+If `delete` commits after `rewrite`:
+
+- if `delete.3_2` is a position delete, `delete` has to be retried from beginning because the old file is removed.
+- if `delete.3_2` is an equality delete, `delete` can succeed after retry because `delete.3_2` will gain a higher sequence number `102` and be applied to `data.5` with sequence number `101`.
+
+If `rewrite` commits after `delete`:
+
+- if `delete.3_2` is a position delete, `rewrite` has to be retried from beginning because it removes `data.3` which undeletes `delete.3_2`.
+- if `delete.3_2` is an equality delete, `rewrite` can commit `data.5` with sequence number `100`, so `delete.3_2` with sequence number `101` can still be applied to `data.5`.
+
+From the analysis above, we can conclude that:
+
+1. equality delete is faster to write, but requires more effort in maintenance to have as good read performance as position delete.
+2. if there are lots of data file removals and row-level deletes commits happening at the same time, equality delete could ensure that all the operations are committed without conflict.
+3. if a table does not receive frequent deletes or data file removals and row-level deletes could be isolated, position delete is recommended for better performance.
+
+Similar to the tradeoff between copy-on-write and merge-on-read, the right choice of the delete file to write depends on the specific application requirements.
+For instance, in the change data capture example we described above, we choose to write an equality delete to apply to existing table data because it is too slow to perform a scan and then generate position deletes in a streaming pipeline. 
+However, because we can quickly know the row position of unflushed data, it is more desirable to write a position delete for it.
+
+## Compaction
+
+In Iceberg, we are interested in the following 2 types of compactions:
+
+- **data compaction**: optimization of data files into better layout through algorithms like bin-packing, sorting, z-ordering, etc.
+- **delete compaction**: optimization of delete files into better layout, which also includes removing delete files
+
+!!!Note
+    The big data community commonly uses the terms _major compaction_ and _minor compaction_ to describe compaction features. 
+    However, we notice that using these terms could easily cause confusion because people have different definitions for these terms.
+    Therefore, we do not use these terms in this article, and we will focus on explaining specific Iceberg actions that compact data and delete files.
+
+### Data compaction
+
+Iceberg data compaction is done through the `RewriteDataFiles` action.
+Users can choose different rewrite strategies such as bin-packing or sorting to rewrite files into optimized layout.
+More details could be found in the procedure pages for each engine that implements this action, including Spark and Flink.
```

Comment:
Probably link out "spark and filnk here"

Reply:
added